src/plugins/intel_cpu/tests/unit/graph/merge_transpose_reorder_test.cpp

```diff
@@ -305,4 +306,104 @@ INSTANTIATE_TEST_SUITE_P(smoke_Run_MergeTransposeReorderWithReshape,
                          ::testing::Combine(::testing::ValuesIn(input_shapes_with_reshape),
                                             ::testing::ValuesIn(test_params_with_reshape)));
 
-}  // namespace
\ No newline at end of file
+}  // namespace
+
+TEST(MergeTransposeReorderWithReshape_InplaceConflict, smoke_Run_MergeTransposeReorderWithReshape_InplaceConflict) {
+    /*  Initial graph:
+                  Parameter(Layout: nspc)
+                   /                 \
+                Reshape             Dummy1(Layout: nspc, LOOK_UP INPLACE)
+                   |                   |
+               Transpose             Result1
+                   |
+               Dummy0(LOOK_UP INPLACE)
+                   |
+                Result0
+
+        In CreateGraph():
+            1. Reorder will be inserted between Parameter -> Reshape to convert layout
+            2. Reorder -> Reshape -> Transpose will be merged to Reorder
+            3. Reorder will be inserted between Parameter -> Dummy1 by ResolveComplexInplaceConflicts()
+                   Parameter
+                   /       \
+                Reorder   Reorder
+                   |         |
+                Dummy0    Dummy1
+                   |         |
+                Result0   Result1
+    */
+    Config conf;
+    conf.rtCacheCapacity = 100;
+    auto context = std::make_shared<GraphContext>(conf, nullptr, false);
+
+    std::unique_ptr<Graph> graph = std::unique_ptr<Graph>(new Graph());
+
+    const ov::Shape testShape{1, 128, 128, 128};
+    ov::ParameterVector params{std::make_shared<ov::op::v0::Parameter>(ov::element::Type_t::f32, ov::Shape{1, 128, 128, 128})};
+
+    auto shape_constant = std::make_shared<ov::op::v0::Constant>(ov::element::Type_t::i32, ov::Shape{3}, std::vector<int64_t>{1, 128, 16384});
+    auto reshape_node = std::make_shared<ov::op::v1::Reshape>(params[0], shape_constant, true);
+    auto order_constant = std::make_shared<ov::op::v0::Constant>(ov::element::Type_t::i32, ov::Shape{3}, std::vector<int64_t>{0, 2, 1});
+    auto transpose_node = std::make_shared<ov::op::v1::Transpose>(reshape_node, order_constant);
```

Comment:
There is no point in allocating so much memory. Suggest reducing tensor sizes to `{1, 8, 8, 8}`.

Reply:
done